sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -75,10 +76,15 @@ export function convertRequestForQueryBatch(batch: QueryLogsBatch): GeneratedBat
  * @internal
  */
 export function convertResponseForQueryBatch(
-  generatedResponse: GeneratedQueryBatchResponse
+  generatedResponse: GeneratedQueryBatchResponse,
+  rawResponse: FullOperationResponse
 ): QueryLogsBatchResult {
-  const fixApplied = fixInvalidBatchQueryResponse(generatedResponse);
+  const fixApplied = fixInvalidBatchQueryResponse(generatedResponse, rawResponse);
 
+  /* Sort the ids that are passed in with the queries, as numbers instead of strings
+   * It is not guaranteed that service will return the responses for queries in the same order
+   * as the queries are passed in
+   */
   const newResponse: QueryLogsBatchResult = {
```

Comment:
Just wondering do you know why the newResponse's entries need to be sorted at all? Ideally we'd return them in the same order as the service does

Reply:
it is out of the scope of this PR but following-up on @maorleger's comment, make sure the other languages are doing the same.

Reply:
Yes the newResponse's entries need to be sorted in the id. Bcz the service doesn't return the results in the same order we pass

Reply:
I would suggest discussing that behavior with the v-team. By sorting by ID the SDK is adding an implicit API which someone somewhere would depend on even if it's not documented. 

I think at a minimum it should be discussed with the v-team and made consistent across languages but consider whether this is a behavior we want to support at all or if we should return results in the order the service passes them. Not blocking this PR at all, by the way

Reply:
Discussed offline with Karishma, didn't realize `id` was an internal only thing so this all makes sense. Resolving!